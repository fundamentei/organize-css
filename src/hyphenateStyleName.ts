const uppercasePattern = /([A-Z])/g;
const msPattern = /^ms-/;
const hyphenateCache: Record<string, string> = {};

export default function hyphenateStyleName(styleName: string): string {
  if (hyphenateCache.hasOwnProperty(styleName)) {
    return hyphenateCache[styleName];
  }

  const hyphenatedString = styleName
    .replace(uppercasePattern, "-$1")
    .toLowerCase()
    .replace(msPattern, "-ms-");

  hyphenateCache[styleName] = hyphenatedString;
  return hyphenateCache[styleName];
}
