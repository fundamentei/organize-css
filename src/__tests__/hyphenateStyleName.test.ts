import hyphenateStyleName from "../hyphenateStyleName";

describe("hyphenateStyleName", () => {
  it("hyphenates border properties", () => {
    const properties = [
      "borderBlockColor",
      "borderBlockEndColor",
      "borderBlockEndStyle",
      "borderBlockEndWidth",
      "borderBlockStartColor",
      "borderBlockStartStyle",
      "borderBlockStartWidth",
      "borderBlockStyle",
      "borderBlockWidth",
      "borderBottomColor",
      "borderBottomLeftRadius",
      "borderBottomRightRadius",
      "borderBottomStyle",
      "borderBottomWidth",
      "borderCollapse",
      "borderEndEndRadius",
      "borderEndStartRadius",
      "borderImageOutset",
      "borderImageRepeat",
      "borderImageSlice",
      "borderImageSource",
      "borderImageWidth",
      "borderInlineColor",
      "borderInlineEndColor",
      "borderInlineEndStyle",
      "borderInlineEndWidth",
      "borderInlineStartColor",
      "borderInlineStartStyle",
      "borderInlineStartWidth",
      "borderInlineStyle",
      "borderInlineWidth",
      "borderLeftColor",
      "borderLeftStyle",
      "borderLeftWidth",
      "borderRightColor",
      "borderRightStyle",
      "borderRightWidth",
      "borderSpacing",
      "borderStartEndRadius",
      "borderStartStartRadius",
      "borderTopColor",
      "borderTopLeftRadius",
      "borderTopRightRadius",
      "borderTopStyle",
      "borderTopWidth"
    ];

    expect(properties.map(hyphenateStyleName)).toMatchSnapshot();
  });

  it("hyphenates WebKit CSS properties", () => {
    const properties = [
      "WebkitBoxAlign",
      "WebkitBoxDirection",
      "WebkitBoxFlex",
      "WebkitBoxFlexGroup",
      "WebkitBoxLines",
      "WebkitBoxOrdinalGroup",
      "WebkitBoxOrient",
      "WebkitBoxPack",
      "WebkitScrollSnapPointsX",
      "WebkitScrollSnapPointsY"
    ];

    expect(properties.map(hyphenateStyleName)).toMatchSnapshot();
  });
});
