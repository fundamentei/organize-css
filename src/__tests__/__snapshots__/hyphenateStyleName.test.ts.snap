// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hyphenateStyleName hyphenates WebKit CSS properties 1`] = `
Array [
  "-webkit-box-align",
  "-webkit-box-direction",
  "-webkit-box-flex",
  "-webkit-box-flex-group",
  "-webkit-box-lines",
  "-webkit-box-ordinal-group",
  "-webkit-box-orient",
  "-webkit-box-pack",
  "-webkit-scroll-snap-points-x",
  "-webkit-scroll-snap-points-y",
]
`;

exports[`hyphenateStyleName hyphenates border properties 1`] = `
Array [
  "border-block-color",
  "border-block-end-color",
  "border-block-end-style",
  "border-block-end-width",
  "border-block-start-color",
  "border-block-start-style",
  "border-block-start-width",
  "border-block-style",
  "border-block-width",
  "border-bottom-color",
  "border-bottom-left-radius",
  "border-bottom-right-radius",
  "border-bottom-style",
  "border-bottom-width",
  "border-collapse",
  "border-end-end-radius",
  "border-end-start-radius",
  "border-image-outset",
  "border-image-repeat",
  "border-image-slice",
  "border-image-source",
  "border-image-width",
  "border-inline-color",
  "border-inline-end-color",
  "border-inline-end-style",
  "border-inline-end-width",
  "border-inline-start-color",
  "border-inline-start-style",
  "border-inline-start-width",
  "border-inline-style",
  "border-inline-width",
  "border-left-color",
  "border-left-style",
  "border-left-width",
  "border-right-color",
  "border-right-style",
  "border-right-width",
  "border-spacing",
  "border-start-end-radius",
  "border-start-start-radius",
  "border-top-color",
  "border-top-left-radius",
  "border-top-right-radius",
  "border-top-style",
  "border-top-width",
]
`;
